### ISSUE TITLE
Web developers cant get phonetic runs that are sent from some East Asian IMEs

### ISSUE THREAD

**Issue by eyalg-msft** [w3c/editing#299]
We want to get the phonetic runs in japanese furigana, where the final selected shown output is kanji.
Please expose an API to get the phonetic runs.

**Comment by masayuki-nakano** [comment]
I love the API too, but in my understanding, it's available only on Windows (at least on Desktop) if the IME is IMM.
https://docs.microsoft.com/windows/win32/api/imm/nf-imm-immgetcompositionstringw

However, the document has deleted about the option (the second parameter), and IIRC, TSF does not have the same API. So, I don't think that it's possible on any Desktop OSes.

**Comment by rniwa** [comment]
I don't think this is even possible in general case. For a single sequence of kanji characters, there could be multiple ways to annotate those characters. What happens when the user types in a sequence of characters by simply selecting those characters on UI instead of typing manually for example?

**Comment by xfq** [comment]
Not sure whether it's related to this feature request, but if you type in Chinese, you can find the input codes for characters on macOS: https://support.apple.com/ja-jp/guide/chinese-input-method/cim11846/mac
